app/channels/editor_channel.rb

```diff
@@ -0,0 +1,62 @@
+# The Editor Channel is opened when a user navigates to the edit page of an
+# Evidence, Issue, Note, or Card. The subscription is closed when the user
+# navigates away from the edit page. An open subscription receives updates
+# from the editor and saves them as new revisions of the resource. The channel
+# then broadcasts the resource's updated_at field back to the subscriber so
+# the form can update the original_updated_at field used for conflict
+# resolution. Although multiple users may be subscribed the channel broadcasts
+# messages out with a user scope. This means users will not receive live updates
+# of changes from other users.
+class EditorChannel < ApplicationCable::Channel
+  include ProjectScopedChannels
+  include PaperTrailActivity
+
+  attr_accessor :resource
+
+  AUTOSAVE_EVENT = 'auto-save'.freeze
+
+  def subscribed
+    reject and return unless find_resource(params)
+
+    stream_for [current_user, current_project, resource]
+  end
+
+  def save(params)
+    resource.paper_trail_event = AUTOSAVE_EVENT
+
+    if resource.update_attributes parsed_params(params['data'])
+      self.class.broadcast_to([current_user, current_project, resource], resource.reload.updated_at.to_i)
+    end
+  end
+
+  private
+
+  def find_resource(params)
+    return unless %w[evidence issue note card].include? params['resource_type']
+
+    if params['resource_type'] == 'card'
+      authorized_list_ids = List.where(board_id: current_project.boards.select(:id)).select(:id)
+      @resource ||= Card.find_by(id: params['resource_id'], list_id: authorized_list_ids)
```

Comment:
This looks like 3 queries for a single fetch: board IDs, list IDs, Card

What about (not tested):

```suggestion
      @resource ||= current_project.boards.includes(:list, list: cards).lists.select{ list == params}.cards.select{ card == params}.first
```
Either way it's a big chain, but I'd say less SQL is better than more SQL?

Reply:
Not better.
It's actually still 3 queries
```
  Board Load (0.2ms)  SELECT "boards".* FROM "boards"
  List Load (0.1ms)  SELECT "lists".* FROM "lists" WHERE "lists"."board_id" = ?  [["board_id", 1]]
  Card Load (0.3ms)  SELECT "cards".* FROM "cards" WHERE "cards"."list_id" IN (?, ?)  [["list_id", 1], ["list_id", 2]]
```
And then requires filtering the ruby arrays. The original format is 3 queries and returns the record directly.